### PR TITLE
feat:etcd as configuration center

### PR DESCRIPTION
**Pull request by liu-song**
**What this PR does**:
https://github.com/dubbogo/arana/issues/62
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -194,3 +207,72 @@ func LoadV2(path string) *ConfigMap {
...
+	}
+	return Client, nil
+}
+
+// PutConfigToEtcd initialize local file config into etcd，only be used in when etcd don't hava data.
+func (c *Client) PutConfigToEtcd(configPath string) error {
+	config := LoadV2(configPath)
+	configJson, err := json.Marshal(config)
+	if err != nil {
+		return fmt.Errorf("config json.marshal failed  %v err:", err)
```
Comment (jjeffcaii) [inline pkg/config/config.go:235]: How about use `github.com/pkg/errors` instead? For example: `pkgErrors.Wrapf(err, "....")`

File: pkg/config/config.go [hunk 2]
```diff
@@ -194,3 +207,72 @@ func LoadV2(path string) *ConfigMap {
...
+	}
+
+	return nil
+}
+
+// LoadConfigFromEtcd get key value from etcd
+func (c *Client) LoadConfigFromEtcd(configKey string) (string, error) {
+	resp, err := c.client.Get(configKey)
+	if err != nil {
+		return "", fmt.Errorf("Get remote config fail error %v", err)
```
Comment (jjeffcaii) [inline pkg/config/config.go:265]: Same as above.

File: pkg/config/config.go [hunk 3]
```diff
@@ -194,3 +207,72 @@ func LoadV2(path string) *ConfigMap {
...
+}
+
+func NewClient(endpiont []string) (*Client, error) {
+	tmpClient, err := etcdv3.NewConfigClientWithErr(
+		etcdv3.WithName(etcdv3.RegistryETCDV3Client),
+		etcdv3.WithTimeout(10*time.Second),
+		etcdv3.WithEndpoints(endpiont...),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("Init etcd client fail error %v", err)
```
Comment (jjeffcaii) [inline pkg/config/config.go:222]: How about use github.com/pkg/errors instead? For example: pkgErrors.Wrapf(err, "....")

File: pkg/config/db.go [hunk 4]
```diff
@@ -116,6 +114,32 @@ type (
...
+		Executor      string            `yaml:"executor" json:"executor"`
+	}
+
+	ListenerV2Config struct {
+		ServerVersion string `yaml:"server_version" json:"server_version"`
+		Users         *Users `yaml:"users" json:"users"`
+	}
+
+	Users struct {
+		Dksl string `yaml:"dksl" json:"dksl"`
```
Comment (jjeffcaii) [inline pkg/config/db.go:130]: Why `Dksl`???

**Review by jjeffcaii — COMMENTED** [review]
Pls fix

File: pkg/config/db.go (hunk 4 above)
Comment (liu-song) [reply pkg/config/db.go]: done

File: pkg/config/db.go [hunk 5]
```diff
@@ -116,6 +113,33 @@ type (
...
+
+	ListenerV2Config struct {
+		ServerVersion string `yaml:"server_version" json:"server_version"`
+		Users         *Users `yaml:"users" json:"users"`
+	}
+
+	Users struct {
+		Username string `yaml:"username" json:"username"`
+		Password string `yaml:"password" json:"password"`
+	}
```
Comment (jjeffcaii) [inline pkg/config/db.go:131]: Emm... I think there should be 2 possible patterns:
1. `Users` should be `[]*User`, the struct of `User` is like current `Users`.
2. Or, the `Users` should be `map[string]string`, key means  username and value means password.
@dk-lockdown @AlexStocks @luky116 What do you guys think?

Comment (liu-song) [reply pkg/config/db.go:131]: Get, i think []*User is better.

**Review by AlexStocks — APPROVED** [review]

**Review by jjeffcaii — APPROVED** [review]
LGTM

**Pull request merged**